presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/SqlCustomization.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+/**
+ * Encapsulates strategies to use for various parts of SQL generation in {@link QueryBuilder}.
+ */
+public interface SqlCustomization
+{
+    String quote(String name);
+
+    default List<String> columnNames(List<JdbcColumnHandle> handles)
+    {
+        return handles.stream()
+                .map(JdbcColumnHandle::getColumnName)
+                .map(this::quote)
+                .collect(toImmutableList());
+    }
+
+    default String fromRelation(String catalog, String schema, String table)
```

Comment:
just `relation` or `getRelation`? What is the purpose of the prefix `from`?

Reply:
`quoteRelation` would be consistent

or we change all the terminology here to more abstract terms like "render" or "format" or "toSql"

* this method -> `relationNameToSql`
* `quote` -> `identifierToSql`

Reply:
This is producing the relation for the `FROM` clause. I named it "relation" specifically because it might be producing a subquery or other non-table relation. It's not simply quoting a relation name.

Reply:
I misunderstood the meaning  of "from" in the method name.